build2cmake/src/cmake/utils.cmake

```diff
@@ -86,9 +85,8 @@ function (hipify_sources_target OUT_SRCS NAME ORIG_SRCS)
     BYPRODUCTS ${HIP_SRCS}
     COMMENT "Running hipify on ${NAME} extension source files.")
 
-  set_source_files_properties(
-    ${HIP_SRCS}
-    PROPERTIES INCLUDE_DIRECTORIES ${include_dirs})
+  # Include the source directory and the current directory in the include path.
+  include_directories(${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
```

Comment:
Includes should be specific to files, or at most to a kernel fileset, not global. We should use the other solution where we skip setting the source file property if it is NOTFOUND. Though instead of the conditional, I'd try if something like this works:

```
get_source_file_property(include_dirs "${SRC}" INCLUDE_DIRECTORIES) 
if (NOT include_dirs)
  set(include_dirs "")
endif()
```

(`NOTFOUND` is a falsy value in CMake, so it should not be necessary to explicitly match for it.)

Reply:
this makes sense! thanks for the info and snippet, I've updated the code in the latest commit to use the conditional above and it works without any issues. Thanks 🙏